slimDNS.py

```diff
@@ -291,7 +297,11 @@ def process_packet(self, buf, addr):
         # We fake the handling of unicast replies. If the packet came
         # from the mutlicast port we multicast the reply but if it
         # came from any other port we unicast the reply.
-        self.sock.sendto(buf[:o], (_MDNS_ADDR, _MDNS_PORT) if addr[0] == _MDNS_PORT else addr)
+        # self.sock.sendto(buf[:o], (_MDNS_ADDR, _MDNS_PORT) if addr[0] == _MDNS_PORT else addr)
+        
+        # XXX Always multicast the reply. Faked handling of unicast replies
+        # seems to cause replies not reaching its destination in some cases.
+        self.sock.sendto(buf[:o], (_MDNS_ADDR, _MDNS_PORT))
```

Comment:
Actually I'm sort of surprised that the old code worked at all, since it looks like I was comparing the source *IP address* to `_MDNS_PORT`. Can you try:
```
self.sock.sendto(buf[:o], (_MDNS_ADDR, _MDNS_PORT) if addr[1] == _MDNS_PORT else addr)
```
and see if it fixes the problem you were having?

Reply:
Changing addr[0] to addr[1] as you suggested fixes the problem I am having.